### ISSUE TITLE
Test Fail on rebase-to-master branch:cctest and mjsunit

### ISSUE THREAD

**Issue by 13nash** [v8-riscv/v8#400]
**Describe the bug**
cctest and mjsunit cases failed and they have the same error log.

1 cctest case: 
cctest/test-heap/Regress10774

11 mjsunit cases:
mjsunit/const-field-tracking-2 
mjsunit/compiler/test-dynamic-map-check-deprecated-maps-polymorphic 
mjsunit/compiler/test-dynamic-map-check-deprecated-maps 
mjsunit/compiler/test-dynamic-map-culibheck-deprecated-maps2 
mjsunit/compiler/test-dynamic-map-check-deprecated-maps3 
mjsunit/compiler/test-dynamic-map-checks-poly-mono 
mjsunit/compiler/test-dynamic-map-checks-wrong-handler 
mjsunit/compiler/test-dynamic-map-checks-wrong-handler1 
mjsunit/compiler/test-dynamic-map-checks 
mjsunit/es6/super-ic-opt-dynamic-map-checks 
mjsunit/regress/regress-1154961 

**To Reproduce**

1. Build with options 
    git chcekout rebase-to-master
    build V8 in debug mode:tools/dev/gm.py riscv64.debug.all 

2. Run test/program 
    tools/run-tests.py -p verbose --report --outdir=out/riscv64.debug cctest 2>&1 |tee cctest.txt 
    tools/run-tests.py -p verbose --report --outdir=out/riscv64.debug mjsunit 2>&1 |tee mjsunit.txt

3. See error
```
--- stderr ---
#
# Fatal error in ../../src/compiler/backend/code-generator.cc, line 171
# Debug check failed: tasm()->SizeOfCodeGeneratedSince(deopt_exit) == expected_offsets[i] + Deoptimizer::kNonLazyDeoptExitSize (24 vs. 28). 
#
#
#
#FailureMessage Object: 0x7fffe6022470
==== C stack trace ===============================

    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/libv8_libbase.so(v8::base::debug::StackTrace::StackTrace()+0x1e) [0x7f8856ff6d2e]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/libv8_libplatform.so(+0x5939d) [0x7f8856f7439d]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/libv8_libbase.so(V8_Fatal(char const*, int, char const*, ...)+0x230) [0x7f8856fdb3e0]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/libv8_libbase.so(+0x40dac) [0x7f8856fdadac]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/libv8_libbase.so(V8_Dcheck(char const*, int, char const*)+0x27) [0x7f8856fdb497]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/libv8.so(v8::internal::compiler::CodeGenerator::AssembleDeoptImmediateArgs(v8::internal::ZoneVector<v8::internal::compiler::ImmediateOperand*> const*, v8::internal::Label*)+0x1f8) [0x7f8856577078]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/libv8.so(v8::internal::compiler::CodeGenerator::AssembleDeoptimizerCall(v8::internal::compiler::DeoptimizationExit*)+0x22b) [0x7f885657755b]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/libv8.so(v8::internal::compiler::CodeGenerator::AssembleCode()+0x104c) [0x7f88565785fc]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/libv8.so(v8::internal::compiler::AssembleCodePhase::Run(v8::internal::compiler::PipelineData*, v8::internal::Zone*)+0x25) [0x7f88569cf395]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/libv8.so(void v8::internal::compiler::PipelineImpl::Run<v8::internal::compiler::AssembleCodePhase>()+0xb3) [0x7f88569c2b73]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/libv8.so(v8::internal::compiler::PipelineImpl::AssembleCode(v8::internal::compiler::Linkage*, std::__Cr::unique_ptr<v8::internal::AssemblerBuffer, std::__Cr::default_delete<v8::internal::AssemblerBuffer> >)+0x117) [0x7f88569b3a67]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/libv8.so(v8::internal::compiler::PipelineCompilationJob::ExecuteJobImpl(v8::internal::RuntimeCallStats*, v8::internal::LocalIsolate*)+0x19c) [0x7f88569b32ac]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/libv8.so(v8::internal::OptimizedCompilationJob::ExecuteJob(v8::internal::RuntimeCallStats*, v8::internal::LocalIsolate*)+0x149) [0x7f88554245c9]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/libv8.so(+0x2615172) [0x7f8855436172]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/libv8.so(+0x2609571) [0x7f885542a571]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/libv8.so(v8::internal::Compiler::CompileOptimized(v8::internal::Handle<v8::internal::JSFunction>, v8::internal::ConcurrencyMode, v8::internal::CodeKind)+0x109) [0x7f885542afe9]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/libv8.so(+0x30a9025) [0x7f8855eca025]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/libv8.so(+0x30a53c7) [0x7f8855ec63c7]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/libv8.so(v8::internal::Runtime_CompileOptimized_NotConcurrent(int, unsigned long*, v8::internal::Isolate*)+0x11d) [0x7f8855ec606d]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/libv8.so(v8::internal::Simulator::SoftwareInterrupt()+0x9eb) [0x7f885631c0bb]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/libv8.so(v8::internal::Simulator::DecodeRVIType()+0xdf9) [0x7f88563230f9]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/libv8.so(v8::internal::Simulator::InstructionDecode(v8::internal::Instruction*)+0x1ec) [0x7f885631926c]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/libv8.so(v8::internal::Simulator::Execute()+0xaf) [0x7f8856324ddf]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/libv8.so(v8::internal::Simulator::CallInternal(unsigned long)+0x2d7) [0x7f88563250d7]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/libv8.so(v8::internal::Simulator::CallImpl(unsigned long, int, long const*)+0x397) [0x7f8856325ab7]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/libv8.so(unsigned long v8::internal::SimulatorBase::VariadicCall<unsigned long, v8::internal::Simulator, long (v8::internal::Simulator::*)(unsigned long, int, long const*), unsigned long, unsigned long, unsigned long, unsigned long, long, unsigned long**>(v8::internal::Simulator*, long (v8::internal::Simulator::*)(unsigned long, int, long const*), unsigned long, unsigned long, unsigned long, unsigned long, unsigned long, long, unsigned long**)+0x17d) [0x7f88555cc6ad]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/libv8.so(unsigned long v8::internal::Simulator::Call<unsigned long, unsigned long, unsigned long, unsigned long, unsigned long, long, unsigned long**>(unsigned long, unsigned long, unsigned long, unsigned long, unsigned long, long, unsigned long**)+0x7c) [0x7f88555cc51c]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/libv8.so(v8::internal::GeneratedCode<unsigned long, unsigned long, unsigned long, unsigned long, unsigned long, long, unsigned long**>::Call(unsigned long, unsigned long, unsigned long, unsigned long, long, unsigned long**)+0x65) [0x7f88555cc0a5]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/libv8.so(+0x27a89b1) [0x7f88555c99b1]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/libv8.so(v8::internal::Execution::Call(v8::internal::Isolate*, v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::internal::Object>, int, v8::internal::Handle<v8::internal::Object>*)+0x8b) [0x7f88555c8bfb]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/libv8.so(v8::Script::Run(v8::Local<v8::Context>)+0x3e8) [0x7f88552052c8]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/d8(v8::Shell::ExecuteString(v8::Isolate*, v8::Local<v8::String>, v8::Local<v8::Value>, v8::Shell::PrintResult, v8::Shell::ReportExceptions, v8::Shell::ProcessMessageQueue)+0x76c) [0x560c69265dbc]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/d8(v8::SourceGroup::Execute(v8::Isolate*)+0x3f7) [0x560c69274277]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/d8(v8::Shell::RunMain(v8::Isolate*, bool)+0x1f2) [0x560c69277592]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/d8(v8::Shell::Main(int, char**)+0x10df) [0x560c69278dff]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/d8(main+0x22) [0x560c692792e2]
    /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xe7) [0x7f88520f5b97]
    /home/liangbin/V8_ROOT/v8/out/riscv64.debug/d8(_start+0x2a) [0x560c6924d26a]
Received signal 6
Aborted

**Expected behavior**
These cases pass.

**Errors/Logs**
If applicable, copy/paste the errors or log messages here.

**Additional context**
Add any other context about the problem here.

**Comment by fw-brice** [comment]
I believe this is fixed with Yahan's fix in #399. I am running all of the tests now.

**Comment by fw-brice** [comment]
Yes, I can confirm that this is fixed now. There is only one remaining failure, for which I opened #401.
